src/components/RegisterPage/RegisterForm.tsx

```diff
@@ -43,21 +53,15 @@ export const RegisterForm = ({ onSuccess }: RegisterFormProps) => {
   const [consentDefinitions, setConsentDefinitions] = useState<ConsentType[]>(
     []
   );
-  const [timeoutId, setTimeoutId] = useState<ReturnType<typeof setTimeout>>();
   const [processing, setProcessing] = useState(false);
-  const [disableActionButton, setDisableActionButton] = useState(false);
 
   const { t } = useTranslation();
   const { publisherId } = useAppSelector(selectPublisherConfig);
   const { error: publisherConsentsError } = useAppSelector(
     selectPublisherConsents
   );
 
-  useEffect(() => {
-    return () => {
-      if (timeoutId) clearTimeout(timeoutId);
-    };
-  }, []);
+  const recaptchaRef = useRef<ReCAPTCHA>(null);
 
   const handleClickShowPassword = () =>
```

Comment:
i think all of the handlers defined in this file are good candidates for extraction to a custom hook. something like `useRegisterForm` - it could return the functions, state, etc.

right now this component is very long and could use some refactoring, at least slight